include/boost/math/distributions/non_central_chi_squared.hpp

```diff
@@ -838,7 +838,9 @@ namespace boost
             &r,
             Policy()))
                return (RealType)r;
-         return detail::generic_find_mode(dist, 1 + k, function);
+         bool asymptotic_mode = k < l/4;
```

Comment:
Did you mean `(k/l) < RealType(1)/4`? Your paper mentions using (k/l) < 1/4 as the conservative threshold. As written `asymptotic_mode` will always evaluate to false, and the ternary operator will always evaluate `starting_point = 1 + k` which is what the original code had.

Reply:
Hi @mborland, thank you for your attention here!

Well, mathematically `k < l/4` and `k/l < 1/4` is the same :) Mainly, I tried to avoid dividing by the parameter, because it might turn out to be zero.

Could you please elaborate a bit more on why would the asymptotic mode always evaluate to `false`? Is it something to do with `RealType` divided by `int`?

Reply:
You are correct, and can ignore my comment. I read as `k < 1/4` not `k < l/4`.